### PR TITLE
Add gpu monitor

### PR DESCRIPTION
**Pull request by suzhoum**
Issue #, if available:

Description of changes:

Sends `GPUUtilization` with `instanceID` to `Custom/EC2` metrics on CloudWatch.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/autogluon/bench/cloud/aws/batch_stack/stack.py [hunk 1]
```diff
@@ -146,6 +146,13 @@ def __init__(self, scope: Construct, id: str, static_stack: StaticResourceStack,
             vpc=vpc,
         )
 
+        # Add inbound rule for ssh access
```
Comment (yinweisu) [inline src/autogluon/bench/cloud/aws/batch_stack/stack.py:149]: Shall we remove this?

File: src/autogluon/bench/cloud/aws/batch_stack/stack.py [hunk 2]
```diff
@@ -228,6 +251,7 @@ def __init__(self, scope: Construct, id: str, static_stack: StaticResourceStack,
                 allocation_strategy=batch.AllocationStrategy.BEST_FIT_PROGRESSIVE,
                 vpc=vpc,
                 vpc_subnets=ec2.SubnetSelection(subnets=vpc.private_subnets),
+                # vpc_subnets=ec2.SubnetSelection(subnets=vpc.public_subnets),  # use public subnet for ssh
```
Comment (yinweisu) [inline src/autogluon/bench/cloud/aws/batch_stack/stack.py:254]: ditto

File: src/autogluon/bench/cloud/aws/docker/entrypoint.sh [hunk 3]
```diff
@@ -1,5 +1,7 @@
 #!/bin/bash
 
+service cron start
```
Comment (yinweisu) [inline src/autogluon/bench/cloud/aws/docker/entrypoint.sh:3]: what does this do? Might be good if we have a comment explaining

File: src/autogluon/bench/cloud/aws/docker/Dockerfile [hunk 4]
```diff
@@ -17,5 +27,9 @@ RUN if [ -n "$AG_BENCH_DEV_URL" ]; then \
         && pip install autogluon.bench==$AG_BENCH_VERSION; \
     fi
 
-RUN chmod +x entrypoint.sh
+RUN chmod +x entrypoint.sh /app/gpu_utilization.sh \
+    && echo "*/5 * * * * CDK_DEPLOY_REGION=/app/gpu_utilization.sh >> /var/log/cron.log 2>&1" > /var/spool/cron/crontabs/root \
```
Comment (yinweisu) [inline src/autogluon/bench/cloud/aws/docker/Dockerfile:31]: So basically we are injecting this into `cron.log`, which will be automatically called to report metrics?

File: src/autogluon/bench/cloud/aws/batch_stack/stack.py (hunk 1 above)
Comment (suzhoum) [reply src/autogluon/bench/cloud/aws/batch_stack/stack.py:149]: I meant to keep this in comment for future use. The plan is to add an ssh access option for testing, but I'm fine to remove it in this PR.

File: src/autogluon/bench/cloud/aws/docker/Dockerfile (hunk 4 above)
Comment (suzhoum) [reply src/autogluon/bench/cloud/aws/docker/Dockerfile:31]: That's right. `service cron start` in the `entrypoint.sh` will start the cron server which runs this line according to the schedule. It automatically reports metrics every 5 minutes.

File: src/autogluon/bench/cloud/aws/batch_stack/stack.py (hunk 1 above)
Comment (yinweisu) [reply src/autogluon/bench/cloud/aws/batch_stack/stack.py:149]: Sure. We can keep it then

File: src/autogluon/bench/cloud/aws/docker/Dockerfile (hunk 4 above)
Comment (suzhoum) [reply src/autogluon/bench/cloud/aws/docker/Dockerfile]: It seems to be a pretty standard way to do such kind of thing according to this thread https://stackoverflow.com/questions/37458287/how-to-run-a-cron-job-inside-a-docker-container

**Review by yinweisu — APPROVED** [review]

**Pull request merged**